internal/pkg/dynamic-config/client.go

```diff
@@ -53,31 +53,13 @@ func (d *DynamicContext) FetchAPIKeyError(apiKey string, clusterID string) error
 	return &errors.UnconfiguredAPISecretError{APIKey: apiKey, ClusterID: clusterID}
 }
 
-func (d *DynamicContext) FetchSchemaRegistryByEnvironmentId(context context.Context, accountId string) (*ccloudv1.SchemaRegistryCluster, error) {
-	existingClusters, err := d.Client.SchemaRegistry.GetSchemaRegistryClusters(context, &ccloudv1.SchemaRegistryCluster{
-		AccountId: accountId,
-		Name:      "account schema-registry",
-	})
+func (d *DynamicContext) FetchSchemaRegistryByEnvironmentId(context context.Context, accountId string) (*srcmv2.SrcmV2Cluster, error) {
```

Comment:
No need for this to be a pointer!
```suggestion
func (d *DynamicContext) FetchSchemaRegistryByEnvironmentId(context context.Context, accountId string) (srcmv2.SrcmV2Cluster, error) {
```